solutions/USACO jan21-juststalling.mdx

```diff
@@ -0,0 +1,59 @@
+---
+id: usaco-1085
+source: USACO Bronze 2021 January
+title: Just Stalling
+author: Ananth Kashyap
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_prob3_bronze_jan21.html)
+
+For this problem, we are going to use permutations. First, we can sort the cows in increasing order based on
+height. Then we can keep track of the number of stalls that each cow is able to fit in. Now, to get the total
+number of possibilities that each cow can use in a permutation, all we have to do is, for each cow, subtract
+off the number of other cows that will be using other stalls which this cow is capable of using. To do this, we
+can use the fact that only cows with heights greater than or equal to this cow are required to use one of the
+stalls that this cow is capable of using. Since we know that our cows are sorted in height increasing order,
+the $i$th cow in our list will have $n-i$ cows with heights greater than or equal to it. So to
+get our final answer, we can initialize a cumulative product which will contain our answer, and for every
+$i$ between 1 and $n$, we just have to multiply this cumulative product by the number of
+stalls that the $i$th cow could possibly use, minus $n-i$.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N^2)$
+
+<LanguageSection>
+
+<PySection>
+
+```python
+n = int(input())
+cows = list(map(int, input().split(" ")))
+stalls = list(map(int, input().split(" ")))
+
+cows.sort()
+stalls.sort()
+
+possible_places = [0] * n
+
+# Create a list containing the number of stalls each cow can use.
+for i in range(n):
+	for j in range(n):
+		if cows[i] <= stalls[j]:
+			possible_places[i] += 1
+			
+# Initialize a cumulative product
+product = 1
+
+# For each cow, find out the number of stalls it can use in a valid permutation
+# Use the formula provided in the explanation
+# Multiply the cumulative product by this value
+for i in range(n - 1, -1, -1):
+	product *= (possible_places[i] - (n - i - 1))
+
+print(product)
```

Comment:
```suggestion
    n = int(input())
    cows = list(map(int, input().split(" ")))
    stalls = list(map(int, input().split(" ")))
    
    cows.sort()
    stalls.sort()
    
    possible_places = [0] * n
    
    # Create a list containing the number of stalls each cow can use.
    for i in range(n):
	    for j in range(n):
		    if cows[i] <= stalls[j]:
			    possible_places[i] += 1
			    
    # Initialize a cumulative product
    product = 1
    
    # For each cow, find out the number of stalls it can use in a valid permutation
    # Use the formula provided in the explanation
    # Multiply the cumulative product by this value
    for i in range(n - 1, -1, -1):
	    product *= (possible_places[i] - (n - i - 1))
    
    print(product)
    ```
    n = int(input())
    cows = list(map(int, input().split(" ")))
    stalls = list(map(int, input().split(" ")))
    
    cows.sort()
    stalls.sort()
    
    possible_places = [0] * n
    
    # Create a list containing the number of stalls each cow can use.
    for i in range(n):
	    for j in range(n):
		    if cows[i] <= stalls[j]:
			    possible_places[i] += 1
			    
    # Initialize a cumulative product
    product = 1
    
    # For each cow, find out the number of stalls it can use in a valid permutation
    # Use the formula provided in the explanation
    # Multiply the cumulative product by this value
    for i in range(n - 1, -1, -1):
	    product *= (possible_places[i] - (n - i - 1))
    
    print(product)